import { AppError } from "../utils/error/AppError.js";

export const sendProvideUsernameError = (res) => {
  throw new AppError(422, ["Please provide username ðŸ™‚"]);
};

export const sendProvideEmailError = (res) => {
  throw new AppError(422, ["Please provide user's email ðŸ™‚"]);
};

export const sendProvideUsernameAndPasswordError = (res) => {
  throw new AppError(400, ["Please provide username and password ðŸ˜ž"]);
};

export const sendProvideUsernameAndOtpError = (res) => {
  throw new AppError(422, ["Please provide username and OTP ðŸ™‚"]);
};

export const sendUserAccountNotAvailableError = (res) => {
  throw new AppError(401, [
    "You don't have an account on this platform ðŸ™‚, please proceed to signup",
  ]);
};

export const sendInvalidOtpError = (res) => {
  throw new AppError(401, ["Invalid OTP or OTP has expired ðŸ˜”"]);
};

export const sendInvalidLoginCredentialsError = (res) => {
  throw new AppError(400, ["Invalid login credentials ðŸ™‚"]);
};

export const sendUserAccountNotActiveError = (res) => {
  throw new AppError(401, [
    "Hello! please proceed to activate your account or contact admin for assistance ðŸ™‚",
  ]);
};

export const sendRequestCouldNotBeCompletedError = (res) => {
  throw new AppError(400, [
    "Request could not be completed ðŸ˜”, please try again!",
  ]);
};
